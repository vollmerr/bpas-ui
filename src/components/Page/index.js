import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../theme';

import Header from '../Header';
import Container from '../Container';
import Footer from '../Footer';

const Content = styled(Container)`
  min-height: ${theme.size.contentHeight}px;
`;

function Page({
  children,
  title,
  links,
}) {
  const headerProps = {
    title,
    links,
  };

  return (
    <div>
      <Header {...headerProps} />
      <Content>
        {children}
      </Content>
      <Footer />
    </div>
  );
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
};

export default Page;
