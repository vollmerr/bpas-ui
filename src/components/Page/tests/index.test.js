import React from 'react';
import { shallow } from 'enzyme';
import Page from '../';

const Child = () => <div>child component</div>;
const testTitle = 'test title';

describe('<Page />', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <Page title={testTitle}>
        <Child />
      </Page>,
      { context: { history: {} } }
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders its children', () => {
    const wrapper = shallow(
      <Page title={testTitle}>
        <Child />
      </Page>,
      { context: { history: {} } }
    );
    expect(wrapper.contains(<Child />)).toEqual(true);
  });

  xit('passes the correct props down', () => { });
});
