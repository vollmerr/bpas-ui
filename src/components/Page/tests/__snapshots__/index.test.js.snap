// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders correctly 1`] = `
<div>
  <Header
    title="test title"
  />
  <Styled(Container)>
    <Child />
  </Styled(Container)>
  <Footer />
</div>
`;
